### ISSUE TITLE
:GoRun doesn't work when .c file file exists in the same directory 

### ISSUE THREAD

**Issue by strogiyotec** [fatih/vim-go#3134]
### What did you do? (required: The issue will be **closed** when not provided)
I created a file vim hello.go and it made a default vim-go template . then I run `:GoRun` and got this error 
```
:!'go' 'run' 'package .: C source files not allowed when not using cgo or SWIG: hello.c'  2>&1| tee /tmp/nvim8fD1ZB/4
cannot find package "package .: C source files not allowed when not using cgo or SWIG: hello.c"
```
If I try to do it within a terminal `go run hello.go` it works 

### What did you expect to happen?
I expect vim to print default message vim-go
### What happened instead?
An error described above 
### Configuration (**MUST** fill this out):
		"highlight functions
		let g:go_highlight_functions = 1
		"Highlight extra types
		let g:go_highlight_extra_types = 1
		let g:go_highlight_types = 0
		"when cursor in function it will highligh args for func
		let g:go_auto_type_info = 1
		"enable :gorename
		"		let g:go_rename_command = 'gopls'
		"Highlight function
		let g:go_highlight_function_calls = 1
		"gd will be handled my coc
		let g:go_def_mapping_enabled = 0
		"Highlight operators
		let g:go_highlight_operators = 1
     	let g:go_highlight_debug = 1

#### vim-go version:
master latest 
####  `vimrc` you used to reproduce:

<details><summary>vimrc</summary><br><pre>

</pre></details>

#### Vim version (first three lines from `:version`):

NVIM v0.4.3
Build type: Release
LuaJIT 2.1.0-beta3

#### Go version (`go version`):

go version go1.15.6 linux/amd64

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>

GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/strogiyotec/.cache/go-build"
GOENV="/home/strogiyotec/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/strogiyotec/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/strogiyotec/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build775199053=/tmp/go-build -gno-record-gcc-switches"

</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>

golang.org/x/tools/gopls v0.6.2
    golang.org/x/tools/gopls@v0.6.2 h1:c3zVKbg4zEOayCwgV/RCA3ucNriOyzfTv22fBW+D1EQ=

</pre></details>

#### vim-go configuration:
<details><summary>vim-go configuration</summary><br><pre>
g:go_auto_type_info = 1
g:go_highlight_debug = 1
g:go_highlight_functions = 1
g:go_highlight_extra_types = 1
g:go_highlight_operators = 1
g:go_highlight_function_calls = 1
g:go_def_mapping_enabled = 0
g:go_highlight_types = 0
g:go_jump_to_error = 1
g:go_loaded_gosnippets = 1
g:go_loaded_install = 1
</pre></details>

#### filetype detection configuration:
<details><summary>filetype detection</summary><br><pre>
filetype detection:ON  plugin:ON  indent:ON
</pre></details>

**Comment by bhcleek** [comment]
I'm not sure what's going on with your system, but I can say with certainty that executing `:GoRun` will not produce an error that looks what you've described. It looks like you're trying to run `go run` via the cmd command (see `:help :!`)

**Comment by strogiyotec** [comment]
@bhcleek  thanks for response, I realized one think, in the directory where I have this hello.go file I do have another file called `hello.c`
```
>cat hello.c
#include <stdio.h>

int main()
{
		printf("Hello world");
}
> ls
.  ..  hello.c  hello.go
```
I just checked , If I remove hello.c and run `:GoRun` again then it works. Do you have any idea why is it happening ?

**Comment by bhcleek** [comment]
`:GoRun` runs in the directory. If you don't want it to consider that C file, then you'll need to give provide the file list.e.g. `:GoRun %`

**Comment by bhcleek** [comment]
For clarity, the reaon you're getting seeing `package foo: C source files not allowed when not using cgo or SWIG: main.c
` is because there is a C file and you're not passing the files to use for `:GoRun`, so vim-go has to use `go list` to figure out which files to include and `go list` fails due to the presence of the C file.

I was able to see the error you're getting. That particular form only comes from Neovim; Vim8 shows a different form. My apologies for not recognizing that previously.

**Comment by strogiyotec** [comment]
@bhcleek  I see, Yes If I try to run `go list` in terminal with .c file in the directory I am getting this error as well. I can use :GoRun % and it works as expected. However is it possible to tell :GoRun to skip .c files without passing list of go files to use ? In case If I have two .go files it will be easier just to run :GoRun and assume that command will skip .c files automatically

**Comment by bhcleek** [comment]
Not as far as I know without calling a chain of Vim functions. But that's kind of beside the point. What you're trying to do here generally isn't even supported by `go`: `go build`, `go list`, etc. will all fail on the directory if it contains a C file.

**Comment by strogiyotec** [comment]
I see, Yes I googled this issue and it seems like in version 1.5 they removed c libs and replaced them by go , So I think it's an expected behavior, thanks for your time , should I close the ticket ?

**Comment by bhcleek** [comment]
You can still use C with Go. The changes in 1.5 weren't about the kind of problem you're having: those changes were about how the Go compiler itself is implemented (prior to Go 1.5, the Go compiler was written in C).

But yes, I think we should close this issue.
